src/ui/map.js

```diff
@@ -3074,9 +3073,11 @@ class Map extends Camera {
         if (this.getProjection().name === 'globe') {
             if (this.transform.zoom >= GLOBE_ZOOM_THRESHOLD_MAX) {
                 if (this.transform.projection.name === 'globe') {
+                    this.transform._globeZoomTransition = true;
```

Comment:
As above, changing `transform` state seems unecessary to me. @avpeery @karimnaaji Thoughts on something like the following?
```js
                    this._updateProjection({zoomTransition: true});
```
Alternatively, perhaps we could break `map._updateProjection` into two functions called something like `_changeProjection` and `_transitionGlobeProjection`, which would each call a common function before performing the conditional update?

Reply:
>Alternatively, perhaps we could break map._updateProjection into two functions called something like _changeProjection and _transitionGlobeProjection, which would each call a common function before performing the conditional update?

I agree, splitting things up would make sense. Most of this logic doesn't need to be shared and this function tries to do too much. It's mainly the reason why we have this issue in the first place. I'm not sure about the common function, but I gave it a quick try at separating the concerns of these two different code paths in https://github.com/mapbox/mapbox-gl-js/commit/3aa06d09ca48ff24850cdf548a9665ab4ce8d8d6 as an example to better support this suggestion